### PR TITLE
Mouse DPI awareness (gdigrab)

### PR DESCRIPTION
**Pull request by MeTaXaS4**
We, at Jumpshare, were using gdigrab for screencapture, we played around a bit and realized it wasn't working in high dpi displays, so we fixed it. We hope it will get approved.

### CODE REVIEW

File: libavdevice/gdigrab.c [hunk 1]
```diff
@@ -71,6 +71,9 @@ struct gdigrab {
 
 #define REGION_WND_BORDER 3
 
+    int vertres;
+    int desktopvertres;
```
Comment (TimothyGu) [inline libavdevice/gdigrab.c]: Global variables are not allowed. Put them in a context.

File: libavdevice/gdigrab.c [hunk 2]
```diff
@@ -478,7 +480,11 @@ static void paint_mouse_pointer(AVFormatContext *s1, struct gdigrab *gdigrab)
                 goto icon_error;
             }
         }
-
+		
+		//that would keep the correct location of mouse with hidpi screens
+		pos.x = pos.x * desktopvertres / vertres;
+		pos.y = pos.y * desktopvertres / vertres;
```
Comment (TimothyGu) [inline libavdevice/gdigrab.c]: Wrong indentation

**Comment by NPatch** [comment]
Working with high dpis is always a problem. Would really help if there was a reliable fix.
Thanks man!

**Comment by whispy** [comment]
@TimothyGu was this merged in to master?

**Comment by TimothyGu** [comment]
Actually, I just tested capturing a video on my 150% scaled Windows 8.1 system and the cursor seems to be in the correct position:

```cmd
ffmpeg -f gdigrab -i desktop -crf 23 out.mp4
```

While I did have to manually adjust the display scaling settings for ffplay.exe when playing back, everything seems to work fine for ffmpeg.exe.

**Pull request closed without merging**